src/main/java/org/quantumbadger/redreader/reddit/prepared/RedditPreparedPost.java

```diff
@@ -776,12 +791,12 @@ protected void onCallbackException(final Throwable t) {
 					}
 
 					@Override
-					protected void onFailure(final RequestFailureType type, final Throwable t, final Integer status, final String readableMessage) {
+					protected void onFailure(final @CacheRequest.RequestFailureType int type, final Throwable t, final Integer status, final String readableMessage) {
 						revertOnFailure();
 						if(t != null) t.printStackTrace();
 
 						final RRError error = General.getGeneralErrorForFailure(context, type, t, status,
-								"Reddit API action: " + action.toString() + " " + src.getIdAndType());
+								"Reddit API action code: " + action + " " + src.getIdAndType());
```

Comment:
Yeah I wasn't really sure how to get around that, as though IntDef is lighter it is missing some "features" of enums.